lib/AST/ASTPrinter.cpp

```diff
@@ -3529,6 +3526,9 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
 
     ModuleDecl *M = D->getDeclContext()->getParentModule();
 
+    if (!Options.FullyQualifiedTypesIfAmbiguous && !M->isBuiltinModule())
```

Comment:
I think this is equivalent today, but we could go even further by always returning true if `M->isBuiltinModule()` and then testing `Options.FullyQualifiedTypesIfAmbiguous` after that.

Reply:
No problem, I've separated the two checks